### PR TITLE
mount: PEP8 compliancy and doc fixes

### PR DESCRIPTION
**Pull request by dagwieers**
##### SUMMARY
This PR includes:
- PEP8 compliancy fixes
- Documentation fixes

##### ISSUE TYPE
 - Feature Pull Request
 - Docs Pull Request

##### COMPONENT NAME
mount

##### ANSIBLE VERSION
v2.5

### CODE REVIEW

File: lib/ansible/modules/system/mount.py [hunk 1]
```diff
@@ -76,7 +66,7 @@
         point will be created.
       - Similarly, specifying C(absent) will remove the mount point directory.
     required: true
-    choices: ["present", "absent", "mounted", "unmounted"]
+    choices: [ absent, mounted, present, unmounted ]
```
Comment (jtyr) [inline lib/ansible/modules/system/mount.py:69]: There should be no spaces at the beginning and the end of the square brackets content.

File: lib/ansible/modules/system/mount.py [hunk 2]
```diff
@@ -1,20 +1,18 @@
...
+# Copyright: (c) 2012, Red Hat, inc
 # Written by Seth Vidal
 # based on the mount modules from salt and puppet
 # GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
 
 from __future__ import absolute_import, division, print_function
 __metaclass__ = type
 
-
 ANSIBLE_METADATA = {'metadata_version': '1.1',
```
Comment (jtyr) [inline lib/ansible/modules/system/mount.py:12]: Please format this like this:

```python
ANSIBLE_METADATA = {
    'metadata_version': '1.1',
    'status': ['preview'],
￼    'supported_by': 'core'
}
```

File: lib/ansible/modules/system/mount.py [hunk 3]
```diff
@@ -85,16 +75,14 @@
...
   boot:
-    version_added: 2.2
     description:
       - Determines if the filesystem should be mounted on boot.
       - Only applies to Solaris systems.
-    required: false
-    default: yes
-    choices: ["yes", "no"]
+    type: bool
+    default: 'yes'
```
Comment (jtyr) [inline lib/ansible/modules/system/mount.py:84]: I would user double quotes to keep the same style like for `version_added` from the top of the `DOCUMENTATION` block.

File: lib/ansible/modules/system/mount.py [hunk 4]
```diff
@@ -85,16 +75,14 @@
...
+    version_added: '2.2'
```
Comment (jtyr) [inline lib/ansible/modules/system/mount.py:85]: The same here.

File: lib/ansible/modules/system/mount.py [hunk 5]
```diff
@@ -551,23 +539,21 @@ def get_linux_mounts(module, mntinfo_file="/proc/self/mountinfo"):
...
     module = AnsibleModule(
         argument_spec=dict(
-            boot=dict(default='yes', choices=['yes', 'no']),
-            dump=dict(),
-            fstab=dict(default=None),
-            fstype=dict(),
-            path=dict(required=True, aliases=['name'], type='path'),
-            opts=dict(),
+            boot=dict(type='bool', default=True),
+            dump=dict(type='str'),
```
Comment (jtyr) [inline lib/ansible/modules/system/mount.py:543]: This is not required as `type=str` is the default type. The same bellow.

File: lib/ansible/modules/system/mount.py [hunk 6]
```diff
@@ -551,23 +539,21 @@ def get_linux_mounts(module, mntinfo_file="/proc/self/mountinfo"):
...
-                required=True,
-                choices=['present', 'absent', 'mounted', 'unmounted']),
+            state=dict(type='str', required=True, choices=['absent', 'mounted', 'present', 'unmounted']),
         ),
         supports_check_mode=True,
         required_if=(
             ['state', 'mounted', ['src', 'fstype']],
-            ['state', 'present', ['src', 'fstype']]
-        )
+            ['state', 'present', ['src', 'fstype']],
```
Comment (jtyr) [inline lib/ansible/modules/system/mount.py:555]: This is unnecessary change.

**Review by jtyr — CHANGES_REQUESTED** [review]
LGTM. `shipit`

File: lib/ansible/modules/system/mount.py (hunk 2 above)
Comment (dagwieers) [reply lib/ansible/modules/system/mount.py:12]: This block is not maintained by me. Every module has the block defined like this.

File: lib/ansible/modules/system/mount.py (hunk 1 above)
Comment (dagwieers) [reply lib/ansible/modules/system/mount.py:69]: This is YAML, for Python you're right because of PEP8.

File: lib/ansible/modules/system/mount.py (hunk 5 above)
Comment (dagwieers) [reply lib/ansible/modules/system/mount.py:543]: Well, for the type I think it's better to keep it, to make it obvious. In the past the default type was "raw".

File: lib/ansible/modules/system/mount.py (hunk 6 above)
Comment (dagwieers) [reply lib/ansible/modules/system/mount.py:555]: It's mentioned in PEP8. Check all their examples, they all do it that way and for good reason.
So functionally yes, there's no difference, but that's the case for everything in PEP8.
https://www.python.org/dev/peps/pep-0008/#when-to-use-trailing-commas

File: lib/ansible/modules/system/mount.py (hunk 2 above)
Comment (jtyr) [reply lib/ansible/modules/system/mount.py:12]: It has this format because this is what `pprint` produces. But the [documentation](http://docs.ansible.com/ansible/latest/dev_guide/developing_modules_documenting.html#version-1-1-of-the-metadata) is showing different formatting.

File: lib/ansible/modules/system/mount.py (hunk 1 above)
Comment (jtyr) [reply lib/ansible/modules/system/mount.py:69]: I couldn't find any better example:

https://en.wikipedia.org/wiki/YAML#Basic_components

File: lib/ansible/modules/system/mount.py (hunk 5 above)
Comment (jtyr) [reply lib/ansible/modules/system/mount.py:543]: There was multiple changes in the past but you don't make any backward compatibility fixes for that neither so I don't see a reason why to make exceptions for `type='str'` here.

File: lib/ansible/modules/system/mount.py (hunk 2 above)
Comment (dagwieers) [reply lib/ansible/modules/system/mount.py:12]: Well, you are free to change it. But that part of the file was added by the core team as-is. And almost all modules have it like that.

File: lib/ansible/modules/system/mount.py (hunk 5 above)
Comment (dagwieers) [reply lib/ansible/modules/system/mount.py:543]: Strict typing in this case seems favorable.
